pkg/controllers/user/logging/utils/syslog.go

```diff
@@ -0,0 +1,115 @@
+package utils
+
+import (
+	"crypto/tls"
+	"fmt"
+	"log/syslog"
+	"net"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config/dialer"
+)
+
+type syslogTestWrap struct {
+	*v3.SyslogConfig
+}
+
+func (w *syslogTestWrap) TestReachable(dial dialer.Dialer, includeSendTestLog bool) error {
+	//TODO: for udp we can't use cluster dialer now, how to handle in cluster deploy syslog
```

Comment:
because we use cluster dialer for other targets, the cluster dialer is base on TCP, but syslog could use UDP, so I add todo here to find out a way to support send logs to system log pod deployed in cluster.